interface/usergroup/addrbook_edit.php

```diff
@@ -107,157 +107,174 @@ function typeSelect(a) {
- $option_abook_type = $type_sql_row['option_value'];
- // Set up any abook_type specific settings
- if ($option_abook_type == 3) {
-  // Company centric
-  $form_title = invalue('form_director_title');
-  $form_fname = invalue('form_director_fname');
-  $form_lname = invalue('form_director_lname');
-  $form_mname = invalue('form_director_mname');
-  $form_suffix = invalue('form_director_suffix');
- }
- else {
-  // Person centric
-  $form_title = invalue('form_title');
-  $form_fname = invalue('form_fname');
-  $form_lname = invalue('form_lname');
-  $form_mname = invalue('form_mname');
-  $form_suffix = invalue('form_suffix');
- }
-
-  if ($userid) {
-
-   $query = "UPDATE users SET " .
-    "abook_type = "   . invalue('form_abook_type')   . ", " .
-    "title = "        . $form_title                  . ", " .
-    "fname = "        . $form_fname                  . ", " .
-    "lname = "        . $form_lname                  . ", " .
-    "mname = "        . $form_mname                  . ", " .
-    "suffix = "       . $form_suffix                 . ", " .
-    "specialty = "    . invalue('form_specialty')    . ", " .
-    "organization = " . invalue('form_organization') . ", " .
-    "valedictory = "  . invalue('form_valedictory')  . ", " .
-    "assistant = "    . invalue('form_assistant')    . ", " .
-    "federaltaxid = " . invalue('form_federaltaxid') . ", " .
-    "upin = "         . invalue('form_upin')         . ", " .
-    "npi = "          . invalue('form_npi')          . ", " .
-    "taxonomy = "     . invalue('form_taxonomy')     . ", " .
-	  "cpoe = "         . invalue('form_cpoe')         . ", " .
-    "email = "        . invalue('form_email')        . ", " .
-    "email_direct = " . invalue('form_email_direct') . ", " .
-    "url = "          . invalue('form_url')          . ", " .
-    "street = "       . invalue('form_street')       . ", " .
-    "streetb = "      . invalue('form_streetb')      . ", " .
-    "city = "         . invalue('form_city')         . ", " .
-    "state = "        . invalue('form_state')        . ", " .
-    "zip = "          . invalue('form_zip')          . ", " .
-    "street2 = "      . invalue('form_street2')      . ", " .
-    "streetb2 = "     . invalue('form_streetb2')     . ", " .
-    "city2 = "        . invalue('form_city2')        . ", " .
-    "state2 = "       . invalue('form_state2')       . ", " .
-    "zip2 = "         . invalue('form_zip2')         . ", " .
-    "phone = "        . invalue('form_phone')        . ", " .
-    "phonew1 = "      . invalue('form_phonew1')      . ", " .
-    "phonew2 = "      . invalue('form_phonew2')      . ", " .
-    "phonecell = "    . invalue('form_phonecell')    . ", " .
-    "fax = "          . invalue('form_fax')          . ", " .
-    "notes = "        . invalue('form_notes')        . " "  .
-    "WHERE id = '" . add_escape_custom($userid) . "'";
```

Comment:
If this is the case(ie. lost population of filled out items), would it be better to test for these items prior to the submit?